azurerm/resource_arm_application_insights_analytics_item_test.go

```diff
@@ -0,0 +1,282 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+
+	"github.com/Azure/azure-sdk-for-go/services/appinsights/mgmt/2015-05-01/insights"
+)
+
+func TestAccAzureRMApplicationInsightsAnalyticsItem_basic(t *testing.T) {
+	resourceName := "azurerm_application_insights_analytics_item.test"
+	ri := tf.AccRandTimeInt()
+	config := testAccAzureRMApplicationInsightsAnalyticsItem_basic(ri, testLocation())
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMApplicationInsightAnalyticsItemDestroy("testquery", insights.ItemScopeShared, insights.ItemTypeParameterQuery),
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMApplicationInsightsAnalyticsItemExists(resourceName, "testquery", insights.ItemScopeShared, insights.ItemTypeParameterQuery),
+					resource.TestCheckResourceAttr(resourceName, "name", "testquery"),
+					resource.TestCheckResourceAttr(resourceName, "scope", "shared"),
+					resource.TestCheckResourceAttr(resourceName, "type", "query"),
+					resource.TestCheckResourceAttr(resourceName, "content", "requests #test"),
+				),
+			},
```

Comment:
can we add an import check here?

```suggestion
			},
			{
				ResourceName:      resourceName,
				ImportState:       true,
				ImportStateVerify: true,
			},
```

Reply:
done